Source/CarthageKit/InputFilesInferrer.swift

```diff
@@ -0,0 +1,148 @@
+
+import Foundation
+import Result
+import ReactiveSwift
+import Tentacle
+import XCDBLD
+import ReactiveTask
+
+public final class InputFilesInferrer {
+    
+    typealias LinkedFrameworksResolver = (URL) -> Result<[String], CarthageError>
+    
+    var executableResolver: (URL) -> URL? = { Bundle(url: $0)?.executableURL }
+    
+    private let builtFrameworks: SignalProducer<URL, CarthageError>
+    private let linkedFrameworksResolver: LinkedFrameworksResolver
+    
+    init(builtFrameworks: SignalProducer<URL, CarthageError>, linkedFrameworksResolver: @escaping LinkedFrameworksResolver) {
+        self.builtFrameworks = builtFrameworks
+        self.linkedFrameworksResolver = linkedFrameworksResolver
+    }
+
+    public convenience init(projectDirectory: URL, platform: Platform) {
+        let searchDirectory = projectDirectory.appendingPathComponent(platform.relativePath, isDirectory: true)
```

Comment:
Can you remind me what URL does this resolves to?

Reply:
@blender this one uses path to the `.xcodeproj` file and appends to it relative path for the given platform: i.e. for iOS - `Carthage/Build/iOS`